make_docs.nu

```diff
@@ -12,6 +12,21 @@ def safe-path [] {
   $in | str replace --all '\?' '' | str replace --all ' ' '_'
 }
 
+# NOTE: You should have nushell source code directory sit along with the docs' directory to make it work
+# Get commands in `extra` crate by parsing the source code
+def get-extra-cmds [] {
+    if ('../nushell/crates/nu-cmd-extra/src' | path exists) {
+        cd ..
+        glob nushell/crates/nu-cmd-extra/src/**/*.rs
+            | each { $in | open -r | parse -r 'fn name\(&self\) -> &str \{\n\s+"(?P<name>[^"]+)"\n\s+\}' }
```

Comment:
I was just playing around with this regex and saw that it does not work in Windows. You'd have to have something more like this:

```nushell
glob crates/nu-cmd-extra/src/**/*.rs | 
  each { 
    $in | 
    open -r | 
    str replace '"' '' -a | 
    parse -r 'fn name\(&self\) -> &str \{[\r|\n]\s+(?P<name>.+)[\r|\n]\s+\}'
  } | flatten
```
I think this would work cross-platform but I haven't been able to test it in mac or linux.

Not saying you need to change it if it already works where you run it.

Reply:
Thanks, I will check it tomorrow

Reply:
I have just updated the regex of `parse`, it works on windows and mac.

> I'm kind of wondering if it's better to grab all the commands this way so we don't have to worry about which platform the script was ran on.

`parse` can help to grab all the commands' name, however, to create the docs we need signatures, input/output, examples, categories, etc. I'm afraid it won't work, at least the regexes would be difficult to maintain